File: imc.rb
#Calculo del indice de masa corporal

print "--> Ingrese su estatura (cm): " 
estatura = Integer(gets.chomp)

print "--> Ingrese su peso (kg): " 
peso = Float(gets.chomp)

imc = peso / (Float(estatura)/100)**2 

print "\n---------------------------------------------\n"

if imc <= 15
    print "Su IMC es: ", imc.round(2), " - Severamente bajo peso\n" 
elsif imc > 15 && imc <= 16 
    print "Su IMC es: ", imc.round(2), " - Muy bajo peso\n"
elsif imc > 16 && imc <= 18.5 
    print "Su IMC es: ", imc.round(2), " - Bajo peso\n"
elsif imc > 18.5 && imc <= 25 
    print "Su IMC es: ", imc.round(2), " - Peso saludable\n"
elsif imc > 25 && imc <= 30 
    print "Su IMC es: ", imc.round(2), " - Sobrepeso\n"
elsif imc > 30 && imc <= 35 
    print "Su IMC es: ", imc.round(2), " - Moderadamente obeso\n"
elsif imc > 35 && imc <= 40 
    print "Su IMC es: ", imc.round(2), " - Severamente obeso\n"
elsif imc > 40 && imc <= 45 
    print "Su IMC es: ", imc.round(2), " - Muy severamente obeso\n"
elsif imc > 45 && imc <= 50 
    print "Su IMC es: ", imc.round(2), " - Mórbidamente obeso\n"
elsif imc > 50 && imc <= 60 
    print "Su IMC es: ", imc.round(2), " - Super obeso\n"
elsif imc > 60 
    print "Su IMC es: ", imc.round(2), " - Hiper obeso\n"
end 

print "---------------------------------------------\n"

File: imc.clj
;;Funcion para calcular el indice de masa corporal
(defn imc [peso altura] 
  (/ peso (* altura altura)))

;;Funcion principal del programa
(defn test-imc []
  (println "Ingrese su peso en kg: ") 
  (let [peso (read-string (read-line))]
       
  (println "Ingrese su altura en m: ") 
  (let [altura (read-string (read-line))] 
  
  (let [valor (imc peso altura)]
    ;;(println valor)
    (cond (< valor 15)
      (println "Tu IMC es " valor "- Severamente bajo peso") 
      (< valor 16)
      (println "Tu IMC es " valor "- Muy bajo peso")
      (< valor 18.5)
      (println "Tu IMC es " valor "- Bajo peso")
      (< valor 25)
      (println "Tu IMC es " valor "- Peso saludable")
      (< valor 30)
      (println "Tu IMC es " valor "- Sobrepeso")
      (< valor 35)
      (println "Tu IMC es " valor "- Moderadamente obeso")
      (< valor 40)
      (println "Tu IMC es " valor "- Severamente obeso")
      (< valor 45)
      (println "Tu IMC es " valor "- Muy severamente obeso")
      (< valor 50)
      (println "Tu IMC es " valor "- Morbidamente obeso")
      (< valor 60)
      (println "Tu IMC es " valor "- Super obeso")
      (> valor 60)
      (println "Tu IMC es " valor "- Hiper obeso")
      :else (println "Intente otra vez"))))))
  
  (test-imc)